baseband/vlbi_base/base.py

```diff
@@ -129,15 +129,39 @@ def __init__(self, fh_raw, header0, nchan, bps, complex_data, thread_ids,
                  sample_rate=None):
 
         if frames_per_second is None and sample_rate is None:
-            frames_per_second = self._get_frame_rate(fh_raw, type(header0))
+            try:
+                frames_per_second = self._get_frame_rate(fh_raw,
+                                                         type(header0))
+            except Exception as exc:
+                exc.args += ("_get_frame_rate raised {e.__class__.__name__} "
+                             "error before reading one second of data.  File "
+                             "may have less than one second of data, in "
+                             "which case framerate must manually be passed "
+                             "to file opener.  File could also be "
+                             "corrupt.".format(e=exc),)
+                raise
```

Comment:
I think about this perhaps a bit differently: the user entered `frames_per_second=None`, and it turns out the auto-detection didn't work, so this a different value should be passed in -> this is a `ValueError` independent of why the auto-detection didn't work. On trying a simple example, I also note that the original error still gets reported even if we ignore it here, so I guess there is no need to be too explicit, and one could just do something like:
```
try:
    ....
except Exception:
    raise ValueError("the frame rate could not be auto-detected. "
                     "This can happen if the file has less than one second of data or is corrupted. "
                     "Trying passing in an explicit 'frames_per_second".)
```